### ISSUE TITLE
`keylime_policy` option `--rootfs` results in policy with a mix of absolute and relative paths

### ISSUE THREAD

**Issue by ansasaki** [keylime/keylime#1611]
Is your issue a feature request? If so, please raise it as an [enhancement](https://github.com/keylime/enhancements)

# Environment
 * OS / version: N/A
 * Processor architecture: N/A 
 * TPM Manufacturer: N/A
 * Keylime version: current master (dc118c780a06cb72725f72bc8100665a07a5a12f)

# Description 
When using the `--rootfs` option to create a policy using a directory as the root directory, the resulting policy contains a mix of absolute and relative paths.

## Expected behavior vs. actual behavior 

Running the command `keylime_policy create runtime --rootfs rootfs` where rootfs has the following structure:
```
rootfs/
├── nested
│   └── nested
└── test
```

Actual:
Results in a json similar to:
```
{
  "meta": {
    "version": 1,
    "generator": 3,
    "timestamp": "2024-08-08 08:57:19.017184"
  },
  "release": 0,
  "digests": {
    "rootfs/test": [
      "a948904f2f0f479b8f8197694b30184b0d2ed1c1cd2a1ec0fb85d299a192a447"
    ],
    "/var/tmp/tmt/run-014/plans/upstream-keylime-all-tests/discover/default-0/tests/functional/keylime_policy-commands/rootfs/nested/nested": [
      "414f8e9fd34ff68f66cbdab5ec63a5e738aa107f3454fa7edb51f49528abf9c6"
    ]
  },
  "excludes": [],
  "keyrings": {},
  "ima": {
    "ignored_keyrings": [],
    "log_hash_alg": "sha256",
    "dm_policy": null
  },
  "ima-buf": {},
  "verification-keys": ""
}
```

Expected:

The paths should have only absolute paths and all with the prefix directory `/rootfs` trimmed:
```
{
  "meta": {
    "version": 1,
    "generator": 3,
    "timestamp": "2024-08-08 08:57:19.017184"
  },
  "release": 0,
  "digests": {
    "/test": [
      "a948904f2f0f479b8f8197694b30184b0d2ed1c1cd2a1ec0fb85d299a192a447"
    ],
    "/nested/nested": [
      "414f8e9fd34ff68f66cbdab5ec63a5e738aa107f3454fa7edb51f49528abf9c6"
    ]
  },
  "excludes": [],
  "keyrings": {},
  "ima": {
    "ignored_keyrings": [],
    "log_hash_alg": "sha256",
    "dm_policy": null
  },
  "ima-buf": {},
  "verification-keys": ""
}
```

## Steps to reproduce problem
1. Run `keylime_policy create runtime --rootfs rootfs`

# Relevant logs

Attach any relevant log files that can help to debug your issue.

**Comment by marcostork** [comment]
@ansasaki, I have a fix for the absolute/relative path issue, but I have a couple of questions if you don't mind:

1 - Would it be a problem to change the _remove_prefix_ default value to **True** in _path_digests_()? It doesn't conflict with the current create_runtime_policy as of now.
2 - The code uses .startswith when searching for directories to exclude (for example, it will exclude a "devel" directory). Is there a specific reason for this approach?
3- I found this description for the "log_hash_alg" field (as shown in your example) in Keylime docs: 
"description": "The hash algorithm used for the running hash in IMA entries (second value). The kernel currently hardcodes it to sha1." 
Does this still hold true? I couldn't get any policy to work on my setup before hardcoding the field to "sha1".

**Comment by ansasaki** [comment]
> @ansasaki, I have a fix for the absolute/relative path issue, but I have a couple of questions if you don't mind:
> 
> 1 - Would it be a problem to change the _remove_prefix_ default value to **True** in _path_digests_()? It doesn't conflict with the current create_runtime_policy as of now. 2 - The code uses .startswith when searching for directories to exclude (for example, it will exclude a "devel" directory). Is there a specific reason for this approach aside from excluding the inside directories (/dev/cpu for example)?

@sergio-correia Could you please answer these?

**Comment by sergio-correia** [comment]
> @ansasaki, I have a fix for the absolute/relative path issue, but I have a couple of questions if you don't mind:
> 
> 1 - Would it be a problem to change the _remove_prefix_ default value to **True** in _path_digests_()? It doesn't conflict with the current create_runtime_policy as of now. 2 - The code uses .startswith when searching for directories to exclude (for example, it will exclude a "devel" directory). Is there a specific reason for this approach aside from excluding the inside directories (/dev/cpu for example)?

1. I don't see an issue with that. The obvious use case for removing the prefix was for when giving the path to the extracted initrd
2. That was probably an oversight and ought to be fixed :)

**Comment by marcostork** [comment]
@sergio-correia and @ansasaki, while considering the excluded directories, I see two options:
Option 1 - You won't accept using these directories as rootfs, and you will remove these directories from the rootfs, meaning you won't accept /dev as your rootfs (digests will be `{}`), and you will ignore /dev inside your rootfs.
Option 2 - Option 1 applies, but you also won't accept a rootfs inside one of the excluded directories, meaning you also won't accept /dev/rootfs or /media/rootfs.
The unittest demands the second option; I want to ensure that's the real objective.

**Comment by marcostork** [comment]
Another point about unittest:
https://github.com/marcostork/keylime/blob/ad79ae8af1a70d81059a0d0778a4117425d98742/test/test_create_runtime_policy.py#L310-L311
This will fail when you trim the path. May I fix that?

**Comment by sergio-correia** [comment]
> Another point about unittest: https://github.com/marcostork/keylime/blob/ad79ae8af1a70d81059a0d0778a4117425d98742/test/test_create_runtime_policy.py#L310-L311 This will fail when you trim the path. May I fix it?

Sure, please fix the test as well to conform with the changes you make. About tests, would it be possible to also add a case where it would fail currently (by mixing absolute/relative paths), so that we can be confident that case is also fixed, after your changes?

**Comment by marcostork** [comment]
@sergio-correia, of course, I will include a case for that. About the excluded directories, should we allow rootfs inside an excluded directory? For example, /mnt/rootfs would be a valid rootfs dir considering /mnt is part of `BASE_EXCLUDE_DIRS`?
